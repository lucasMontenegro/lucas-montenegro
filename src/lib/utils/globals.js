import g from "global"
// renaming "global" to "globals"
// declaring a variable named "global" may cause trouble
export default g