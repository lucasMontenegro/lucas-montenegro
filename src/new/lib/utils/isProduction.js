import global from "global"
export default function isProduction () {
  return (
    global.process ? global.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
  )
}