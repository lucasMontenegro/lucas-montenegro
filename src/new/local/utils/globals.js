// renaming "global" to "globals"
// declaring a variable named "global" may cause troubles
exports.default = require("global")