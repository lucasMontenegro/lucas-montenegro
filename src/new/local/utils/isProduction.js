const { default: globals } = require("new/local/utils/globals")
exports.default = function isProduction () {
  return (
    globals.process ? globals.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
  )
}