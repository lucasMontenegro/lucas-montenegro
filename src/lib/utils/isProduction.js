import globals from "lib/utils/globals"
export default function isProduction () {
  return (
    globals.process ? globals.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
    // WebPack will replace this last line with a boolean
  )
}