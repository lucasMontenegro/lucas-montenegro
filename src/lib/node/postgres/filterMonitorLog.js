import path from "path"
import globals from "lib/utils/globals"
import isProduction from "lib/utils/isProduction"
import os from "os"
import fs from "fs"
const logFile = path.resolve(globals.process.cwd(), `pg-errors.log`)
export default function filterMonitorLog (msg, info) {
  if (info.event === `error`) {
    if (!isProduction()) {
      let logText = os.EOL + msg
      if (info.time) {
        logText = os.EOL + logText
      }
      fs.appendFile(logFile, logText, e => {
        if (e) {
          globals.console.error(e)
        }
      })
    }
  } else if (isProduction()) {
    info.display = false
  }
}