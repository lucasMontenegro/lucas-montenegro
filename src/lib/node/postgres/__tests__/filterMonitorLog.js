import globals from "lib/utils/globals"
import isProduction from "lib/utils/isProduction"
import fs from "fs"
import filterMonitorLog from "../filterMonitorLog"
jest.mock(`lib/utils/globals`, () => ({
  __esModule: true,
  default: {
    process: { cwd: () => `/cwd` },
    console: {},
  },
}))
jest.mock(`lib/utils/isProduction`, () => ({ __esModule: true, default: jest.fn() }))
jest.mock(`os`, () => ({
  __esModule: true,
  default: { EOL: `<EOL>` },
}))
jest.mock(`fs`, () => ({
  __esModule: true,
  default: { appendFile: jest.fn() },
}))
describe(`../filterMonitorLog`, () => {
  jestUtils.describeDependencies({
    deps: [
      `lib/utils/globals`,
      `lib/utils/isProduction`,
    ],
    relativeBasePath: __dirname,
  })
  describe(`filterMonitorLog (error event, production true)`, () => {
    it(`should not do anything`, () => {
      isProduction.mockReturnValueOnce(true)
      filterMonitorLog(`msg`, { event: `error` })
      expect(fs.appendFile.mock.calls).toEqual([])
    })
  })
  {
    const cases = [[new Date(`2020-03-09T19:42:12.837Z`)], [undefined]]
    const msg =`filterMonitorLog (error event, production false, info.time %j)`
    describe.each(cases)(msg, time => {
      let errorCallback
      beforeAll(() => {
        isProduction.mockReturnValueOnce(false)
        filterMonitorLog(`msg`, { event: `error`, time })
        try {
          errorCallback = fs.appendFile.mock.calls[0][2]
        } catch (e) {
          if (e.message !== `Cannot read property '2' of undefined`) {
            throw e
          }
        }
      })
      it(`should save the error message to the file system`, () => {
        expect(fs.appendFile.mock.calls).toMatchSnapshot()
        fs.appendFile.mockClear()
      })
      describe(`fs.appendFile error handler (error {})`, () => {
        it(`should log the error`, () => {
          const error = globals.console.error = jest.fn()
          const e = {}
          errorCallback(e)
          delete globals.console.error
          expect(error.mock.calls).toEqual([[{}]])
          expect(error.mock.calls[0][0]).toBe(e)
        })
      })
      describe.each([[null], [undefined]])(`fs.appendFile error handler (error %j)`, e => {
        it(`should not call console.error`, () => {
          const error = globals.console.error = jest.fn()
          errorCallback(e)
          delete globals.console.error
          expect(error.mock.calls).toEqual([])
        })
      })
    })
  }
  describe(`filterMonitorLog (foo event, production true)`, () => {
    it(`should hide the error message`, () => {
      const info = { event: `foo` }
      isProduction.mockReturnValueOnce(true)
      filterMonitorLog(`msg`, info)
      expect(info).toHaveProperty(`display`, false)
      expect(fs.appendFile.mock.calls).toEqual([])
    })
  })
  describe(`filterMonitorLog (foo event, production false)`, () => {
    it(`should hide the error message`, () => {
      const info = { event: `foo` }
      isProduction.mockReturnValueOnce(false)
      filterMonitorLog(`msg`, info)
      expect(info).toEqual({ event: `foo` })
      expect(fs.appendFile.mock.calls).toEqual([])
    })
  })
})