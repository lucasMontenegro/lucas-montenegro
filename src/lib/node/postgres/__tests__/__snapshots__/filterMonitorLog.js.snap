// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../filterMonitorLog filterMonitorLog (error event, production false, info.time "2020-03-09T19:42:12.837Z") should save the error message to the file system 1`] = `
Array [
  Array [
    "/cwd/pg-errors.log",
    "<EOL><EOL>msg",
    [Function],
  ],
]
`;

exports[`../filterMonitorLog filterMonitorLog (error event, production false, info.time undefined) should save the error message to the file system 1`] = `
Array [
  Array [
    "/cwd/pg-errors.log",
    "<EOL>msg",
    [Function],
  ],
]
`;

exports[`../filterMonitorLog local dependencies lib/utils/globals should match the saved hash 1`] = `"658dffcc9a7d7107301d9ee254ba487ada2e2497635e046c509db0828ff3533f"`;

exports[`../filterMonitorLog local dependencies lib/utils/isProduction should match the saved hash 1`] = `"618391bb8e4127638582b2a148593f056858444461170dd8c9dd54252b5ed001"`;
